### PR TITLE
feat(telecom): check otp reference is valid

### PR DESCRIPTION
**Pull request by Steffy29**
| Question         | Answer
| ---------------- | ---
| Branch?          | `master`
| Bug fix?         | yes
| New feature?     | no
| Breaking change? | no
| Tickets          | Fix #UXCT-559
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~Breaking change is mentioned in relevant commits~

## Description

Check the OTP reference set is in valid format.

### CODE REVIEW

**Review by antonymarion — APPROVED** [review]

File: packages/manager/apps/telecom/src/app/telecom/pack/move/building-details/move-building-details.controller.js [hunk 1]
```diff
@@ -75,6 +77,25 @@ export default class MoveBuildingDetailsCtrl {
...
+      /^[A-Z]{2}-[A-Z\d]{4}-[A-Z\d]{4}$/,
+    );
+  }
+
+  /**
+   * Check if PTO reference is valid
+   * @returns boolean
+   */
+  isPtoValid() {
+    return (
```
Comment (antonymarion) [inline packages/manager/apps/telecom/src/app/telecom/pack/move/building-details/move-building-details.controller.js:92]: Maybe factorize this in some helpers (as I am reading twice the same code)

Comment (Steffy29) [reply packages/manager/apps/telecom/src/app/telecom/pack/move/building-details/move-building-details.controller.js:92]: yes I should think about it to do better code into a next feature.

File: packages/manager/apps/telecom/src/app/telecom/pack/migration/building-details/migration-building-details.controller.js [hunk 2]
```diff
@@ -99,6 +101,25 @@ export default class TelecomPackMigrationBuildingDetailsCtrl {
...
+  }
+
+  /**
+   * Check if OTP reference is valid
+   * @returns boolean
+   */
+  isOtpValid() {
+    return (
+      this.model.ptoReference != null &&
+      this.model.ptoReference !== '' &&
```
Comment (ghyenne) [inline packages/manager/apps/telecom/src/app/telecom/pack/migration/building-details/migration-building-details.controller.js:118]: I think your regex manage these case null and ''

File: packages/manager/apps/telecom/src/app/telecom/pack/migration/building-details/migration-building-details.controller.js [hunk 3]
```diff
@@ -35,6 +35,8 @@ export default class TelecomPackMigrationBuildingDetailsCtrl {
       ptoReferenceNotKnown: false,
     };
 
+    this.otpReferenceValid = true;
```
[the remark this answers is not in the document]
Comment (Steffy29) [reply packages/manager/apps/telecom/src/app/telecom/pack/migration/building-details/migration-building-details.controller.js:38]: yes that's what we want, default value is true.

File: packages/manager/apps/telecom/src/app/telecom/pack/migration/building-details/migration-building-details.controller.js (hunk 2 above)
Comment (Steffy29) [reply packages/manager/apps/telecom/src/app/telecom/pack/migration/building-details/migration-building-details.controller.js:118]: no it's not the same need.

File: packages/manager/apps/telecom/src/app/telecom/pack/migration/building-details/migration-building-details.controller.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (ghyenne) [reply packages/manager/apps/telecom/src/app/telecom/pack/migration/building-details/migration-building-details.controller.js:38]: But when you fill the input you use data-ng-change="$ctrl.isOtpRegexValid()"
So this.otpReferenceValid = this.model.ptoReference.match(
      /^[A-Z]{2}-[A-Z\d]{4}-[A-Z\d]{4}$/,
    );
 I think it's no needed.

File: packages/manager/apps/telecom/src/app/telecom/pack/migration/building-details/migration-building-details.controller.js [hunk 4]
```diff
@@ -99,6 +101,13 @@ export default class TelecomPackMigrationBuildingDetailsCtrl {
     }
   }
 
+  /* Check OTP reference value is valid */
+  isOtpRegexValid() {
+    this.otpReferenceValid = this.model.ptoReference.match(
```
Comment (ghyenne) [inline packages/manager/apps/telecom/src/app/telecom/pack/migration/building-details/migration-building-details.controller.js:106]:
```suggestion
    this.otpReferenceValid = this.model.ptoReference?.match(
```
Normally if you are this.model.ptoReference === null or '' you will have false so it's same result for this if statement.

File: packages/manager/apps/telecom/src/app/telecom/pack/migration/building-details/migration-building-details.controller.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (Steffy29) [reply packages/manager/apps/telecom/src/app/telecom/pack/migration/building-details/migration-building-details.controller.js:38]: But if I don't initialize the value by default, the error message about format is displayed.

**Review by ghyenne — APPROVED** [review]

**Review by anooparveti — APPROVED** [review]

**Pull request merged**